src/monotonic/monotonic.jl

```diff
@@ -279,6 +296,33 @@ function calcTangents(::Type{TCoeffs}, x::AbstractVector{<:Number},
     return (m, Δ)
 end
 
+function calcTangents(::Type{TCoeffs}, x::AbstractVector{<:Number},
+    y :: AbstractVector{TEl}, method::AkimaMonotonicInterpolation) where {TCoeffs, TEl}
+
+    # based on Akima (1970),
+    # "A New Method of Interpolation and Smooth Curve Fitting Based on Local Procedures",
+    # doi:10.1145/321607.321609.
+
+    n = length(x)
+    Δ = Vector{TCoeffs}(undef, n-1)
+    m = Vector{TCoeffs}(undef, n)
+    for k in 1:n-1
+        Δk = (y[k+1] - y[k]) / (x[k+1] - x[k])
+        Δ[k] = Δk
+    end
+    Γ = [3Δ[1] - 2Δ[2]; 2Δ[1] - Δ[2]; Δ; 2Δ[n-1] - Δ[n-2]; 3Δ[n-1] - 2Δ[n-2]]
+    for k in 1:n
+        δ = abs(Γ[k+3] - Γ[k+2]) + abs(Γ[k+1] - Γ[k])
+        if δ > zero(δ)
```

Comment:
```suggestion
        if δ ≉ zero(δ)
```

That is `\napprox<tab>`. We should be careful of very small values of `δ` as well. `δ` should be nonnegative.

Reply:
Thanks for pointing me to the right place in the documentation.
I will have a closer look at that one now, too, and then give me thoughts and that suggestion.

Reply:
Alright, I checked out how the `Base.isapprox` function works now.
I figured that due to the statement

> Note that `x ≈ 0` (i.e., comparing to zero with the default tolerances) is equivalent to `x == 0` since the default `atol` is `0`.

we have a situation here where `δ > zero(δ)` is logically and numerically equivalent to `δ ≉ zero(δ)` because we know that  `δ = abs(Γ[k+3] - Γ[k+2]) + abs(Γ[k+1] - Γ[k])` yields a nonnegative value.

---

Therefore I looked at the performance using the `BenchmarkTools` package. Here are the results:

`julia> @benchmark rand() > 0`:

> BenchmarkTools.Trial: 1000000 samples with 1000 evaluations.
>  Range (min … max):  6.000 ns … 333.200 ns  ┊ GC (min … max): 0.00% … 0.00%
>  Time  (median):     6.200 ns               ┊ GC (median):    0.00%
>  Time  (mean ± σ):   6.695 ns ±   1.945 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%
> 
>   █▇▇▄▄   ▄▃  ▁▃   ▁                                  ▄▄ ▄▄▁▂ ▄
>   ██████▇████████▇▇█▇▆▆▅▆▅▅▅▅▅▅▆▅▅▄▅▄▅▅▄▄▄▄▄▄▄▅▄▄▄▃▄▄▆██▅████ █
>   6 ns         Histogram: log(frequency) by time      12.7 ns <
> 
>  Memory estimate: 0 bytes, allocs estimate: 0.

---

> `julia> @benchmark rand() ≉ 0`:
> 
> BenchmarkTools.Trial: 1000000 samples with 1000 evaluations.
>  Range (min … max):  7.100 ns … 172.600 ns  ┊ GC (min … max): 0.00% … 0.00%
>  Time  (median):     7.300 ns               ┊ GC (median):    0.00%
>  Time  (mean ± σ):   8.034 ns ±   2.332 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%
> 
>   █▆▇▄  ▁▃▃  ▂   ▁                          ▄▄▁▅▄ ▂       ▂ ▁ ▄
>   █████▆███████▇▇█▇▅▆▅▅▅▅▅▅▆▅▅▅▅▄▅▄▅▄▅▄▅▄▄▅▆███████▅▇▅█▆█▇█▇█ █
>   7.1 ns       Histogram: log(frequency) by time      15.8 ns <
> 
>  Memory estimate: 0 bytes, allocs estimate: 0.

---

So, looking at the results in terms of performance no change should be made here.
Is there any good reason that is more important than the performance so that we should still change this statement?

Reply:
Have you checked
```!(Γ[k+3] ≈ Γ[k+2] && Γ[k+1] ≈ Γ[k])```

Reply:
Alright,
I created a set of 1_000 random `x` and `y` vectors of length 20 and ran a test function that calls `calcTangents` on all that 1_000 pairs of `(x,y)`. I did that for the following four cases using the same random vectors:
1. `δ > zero(δ)`
2. `!(Γ[k+3] ≈ Γ[k+2] && Γ[k+1] ≈ Γ[k])`
3. `Γ[k+3] ≉ Γ[k+2] && Γ[k+1] ≉ Γ[k]`
4. `Γ[k+3] ≈ Γ[k+2] && Γ[k+1] ≈ Γ[k]` + switching the contents of if and else

Here are the min, median and mean times for 10_000 samples with 1 evaulation of the test function:
1. min:  2.151 ms, median:  2.269 ms, mean:  2.472 ms
2. min:  2.248 ms, median:  2.362 ms, mean:  2.562 ms
3. min:  2.331 ms, median:  2.446 ms, mean:  2.624 ms
4. min:  2.232 ms, median:  2.342 ms, mean:  2.529 ms

We can see that `δ > zero(δ)` is a little superior to all other variants.